src/reducers/assignment.js

```diff
@@ -18,24 +17,51 @@ export function assignment(state = initialState, action) {
           error: false,
           loading: true,
         }
-      }
-    case types.CREATE_CLASSROOM_SUCCESS:
+      };
+    case types.CREATE_ASSIGNMENT_SUCCESS:
       const newlist = state.assignments.data.concat(action.data);
       return { ...state,
         assignments: {
           data: newlist,
           error: false,
           loading: false,
         }
-      }
-    case types.CREATE_CLASSROOM_ERROR:
+      };
+    case types.CREATE_ASSIGNMENT_ERROR:
+      return { ...state,
+        assignments: {
+          data: [],
+          error: action.error,
+          loading: false,
+        }
+      };
+    case types.REQUEST_ASSIGNMENTS:
+      console.log('-'.repeat(40), '\nREQUEST_ASSIGNMENTS\n');
```

Comment:
probably don't need logging, there's a redux middleware that does all this already